### DISCUSSION TITLE
Node updates (@types/node, .nvmrc, etc) misbehaving with default presets

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#17832] [Request Help] [answered]
Noticed earlier today that `@types/node` updates are not being found... I'm not entirely sure why this is the case as I'm 90% certain this was previously working...

Other `node` updates are working as expected, as is other `@types/*` updates

Recreation: https://github.com/setchy/renovate-demo-types-node 
```
"config": {
    "npm": [
      {
        "packageFile": "package.json",
        "deps": [
          {
            "depType": "devDependencies",
            "depName": "@types/node",
            "currentValue": "18.0.0",
            "datasource": "npm",
            "prettyDepType": "devDependency",
            "depIndex": 0,
            "updates": [],
            "warnings": [],
            "versioning": "node",
            "sourceUrl": "https://github.com/DefinitelyTyped/DefinitelyTyped",
            "sourceDirectory": "types/node",
            "homepage": "https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node",
            "currentVersion": "18.0.0",
            "fixedVersion": "18.0.0"
          }
        ],
        "packageJsonName": "renovate-demo-types-node",
        "packageFileVersion": "1.0.0",
        "packageJsonType": "app",
        "managerData": {
          "yarnZeroInstall": false,
          "hasPackageManager": false
        },
        "skipInstalls": true,
        "constraints": {}
      }
    ]
  }
}
```

**Comment by setchy** [comment]
similar issue https://github.com/renovatebot/renovate/discussions/17309, though unclear of how it was resolved...

**Comment by setchy** [comment]
Ok, so, adding the following packageRule to change `@types/node` versioning from `node` to `npm` seems to have done the trick

```
"packageRules": [
    {
        "description": "Reset versioning for @types/node from node to npm",
        "matchPackageNames": [
            "@types/node"
        ],
        "versioning": "npm"
    }
]
 ```
 
 Resulting PR https://github.com/setchy/renovate-demo-types-node/pull/3

**Reply by setchy** [reply]
It may seem like the [workarounds:typesNodeVersioning](https://docs.renovatebot.com/presets-workarounds/#workaroundstypesnodeversioning) may be the root cause, which is part of `workarounds:all` preset used by `config:base`

**Comment by setchy** [comment]
Some further oddities perhaps with `node` updates.

It also appears `.nvmrc` by default uses `versioning: node` and were not suggesting any updates.  After also adding an explicit packageRule as below, it's now being updated in the [PR](https://github.com/setchy/renovate-demo-types-node/pull/3)

```
{
    "description": "experiment with .nvmrc",
    "matchFiles": [
        ".nvmrc"
    ],
    "matchPackageNames": [
        "node"
    ],
    "versioning": "npm"
}
```

That said, now the PR title of `Update Node.js to v18.9.0 ` is misleading since `@types/node` and `node` are being grouped together, although they have different versioning
* `@types/node` - 18.7.18
* node - 18.9.0

**Comment by rarkins** [comment, marked as answer]
It's because 18.x releases are considered unstable and we only update unstable if matching x.y.z

**Reply by setchy** [reply]
Ah ha, thanks @rarkins 

💡 even though v18.x will be a LTS, it's not currently - I hadn't picked up on that important difference.

Created a fresh demo repo https://github.com/setchy/renovate-demo-node-18 and used the following package file which seems to have done the trick
```
"packageRules": [
    {
        "matchPackageNames": [
            "node",
            "@types/node"
        ],
        "ignoreUnstable": false
    }
]
```

**Reply by rarkins** [reply]
Wondering if we can work around this automatically, e.g. with a new workaround which sets ignoreUnstable to be false for `node` packages where matchCurrentVersion = ^18.0.0. The challenge with this is that we have to remember to remove the same workaround right when Node 18 becomes LTS
